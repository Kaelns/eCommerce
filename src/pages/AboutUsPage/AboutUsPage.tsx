import { Box, Icon, List, Stack, Card, CardMedia, CardContent, Link } from '@mui/material';
import { Title } from '@/components/typography/Title/Title';
import { TextBold } from '@/components/typography/TextBold/TextBold';
import { ABOUT_US } from '@/pages/AboutUsPage/AboutUs.constant';
import schoolLogo from '@/assets/rs_school.svg';

export function AboutUsPage(): React.ReactNode {
  return (
    <>
      <Title>About Us Page</Title>
      <Box sx={{ whiteSpace: 'pre-line' }}>
        <TextBold sx={{ textAlign: 'center' }}>
          All roads lead to
          <Link href="https://rs.school/">
            <Icon sx={{ height: 40, width: 100, ml: 20 }}>
              <Box component="img" src={schoolLogo} alt="RS School logo" />
            </Icon>
          </Link>
        </TextBold>
        {ABOUT_US.introduction}
        <List component={Stack} direction={{ xs: 'column', md: 'row' }}>
          {ABOUT_US.members.map((member) => (
            <Card key={member.lastName}>
              <CardMedia>
                <Box component="img" src={member.photo} alt={member.lastName} style={{ width: 200, height: 200 }} />
              </CardMedia>
              <CardContent>
                <Title variant="h6">{`${member.firstName} ${member.lastName}`}</Title>
                <TextBold> Github: </TextBold>
                <Link href={member.github} underline="hover">
                  {member.github}
                </Link>
                <TextBold> Role: </TextBold>
                {member.roles}
                <TextBold> Significant contributions: </TextBold>
                {member.contributions}
                <TextBold> BIO: </TextBold>
                {member.bio}
              </CardContent>
            </Card>
          ))}
        </List>
      </Box>
    </>
  );
}
