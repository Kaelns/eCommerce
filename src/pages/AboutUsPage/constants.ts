export const ABOUT_US = {
  introduction:
    'Hello! We are RadioReactive team.\nWe are a super "serious" radioactive group of programmers. The strength is in the team so we look out for each other, we come to help in any difficult situation. We help to develop as programmers, to correct our bad habits and to learn a lot for further work. Of course, we do not forget about relax and entertainment, where without them. And that\'s all help us to achieve success in this final project.\nAnd next, a little about each of us.',
  members: [
    {
      firstName: 'Yury',
      lastName: 'Ahanouski',
      github: 'https://github.com/kaelns',
      photo: 'https://avatars.githubusercontent.com/u/67708215?v=4',
      roles: 'Team leader',
      contributions:
        "This guy responsible for team collaboration and main visual part of project such as catalog page and product detailed page, navigation and app's header.",
      bio: 'Graduated from Belarusian National Technical University as Energy Engineer. Studied in RS School in 2022 year but was drafted into the army to pay his debt to the homeland and now he is again in the RS School to finish what was started and get more experience and knowledge.'
    },
    {
      firstName: 'Anatoli',
      lastName: 'Zdanevich',
      github: 'https://github.com/zdan3v1ch',
      photo: 'https://avatars.githubusercontent.com/u/84130329?v=4',
      roles: 'The right hand of the leader',
      contributions:
        "This guy is a guru of commercetools API. The average user won't see his work, only the one who familiar with Network tab in Devtools. All API requests are his handiwork.",
      bio: 'Graduated from college, university, and also underwent retraining in “Web Design and Computer Graphics” at the Business Institute of BSU. Completed JS/FE preschool 2022Q2. And now he is again in the RS School in the main course to get more experience and knowledge.'
    },
    {
      firstName: 'Viktoriya',
      lastName: 'Schepalina',
      github: 'https://github.com/90Viktoriya',
      photo: 'https://avatars.githubusercontent.com/u/112850187?v=4',
      roles: 'The left hand of the leader',
      contributions:
        "This girl is responsible for periodic reminders and checks that the work is in full swing. If you have any problems with login, registration or user profile pages, it's also to her.",
      bio: 'Graduated from Surgut State University as specialist in Automated System of information processing and control. Worked as system administrator of SAP system for many years. Forgot all during parental leave. And now she is in RS School to get more experience and knowledge.'
    }
  ]
};
