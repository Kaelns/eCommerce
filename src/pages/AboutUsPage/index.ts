export { AboutUsPage } from '@/pages/AboutUsPage/AboutUsPage';
