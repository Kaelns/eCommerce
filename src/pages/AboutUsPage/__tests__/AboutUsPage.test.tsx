import { AboutUsPage } from '@/pages/AboutUsPage/AboutUsPage';
import { render } from '@testing-library/react';

describe('Given AboutUsPage component: ', () => {
  it('when rendered, should match snapshot', () => {
    const { asFragment } = render(<AboutUsPage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
