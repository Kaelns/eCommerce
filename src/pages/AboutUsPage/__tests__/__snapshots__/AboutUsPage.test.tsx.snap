// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given AboutUsPage component:  > when rendered, should match snapshot 1`] = `
<DocumentFragment>
  <h5
    class="MuiTypography-root MuiTypography-h5 _about-us_3b65ea css-bezmlo-MuiTypography-root"
  >
    About Us Page
  </h5>
  <div
    class="MuiBox-root css-0"
    style="white-space: pre-line;"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
      style="text-align: center;"
    >
      All roads lead to
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
        href="https://rs.school/"
      >
        <span
          aria-hidden="true"
          class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
          style="height: 40px; width: 100px; margin-left: 20px;"
        >
          <img
            alt="RS School logo"
            class="MuiBox-root css-0"
            src="/src/assets/rs_school.svg"
          />
        </span>
      </a>
    </p>
    Hello! We are RadioReactive team.
We are a super "serious" radioactive group of programmers. The strength is in the team so we look out for each other, we come to help in any difficult situation. We help to develop as programmers, to correct our bad habits and to learn a lot for further work. Of course, we do not forget about relax and entertainment, where without them. And that's all help us to achieve success in this final project.
And next, a little about each of us.
    <div
      class="MuiStack-root MuiList-root MuiList-padding css-at4np2-MuiStack-root-MuiList-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
        >
          <img
            alt="Ahanouski"
            src="https://avatars.githubusercontent.com/u/67708215?v=4"
            style="width: 200px; height: 200px;"
          />
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
          >
            Yury Ahanouski
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Github: 
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11iau0r-MuiTypography-root-MuiLink-root"
            href="https://github.com/kaelns"
          >
            https://github.com/kaelns
          </a>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Role: 
          </p>
          Team leader
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Significant contributions: 
          </p>
          This guy responsible for team collaboration and main visual part of project such as catalog page and product detailed page, navigation and app's header.
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             BIO: 
          </p>
          Graduated from Belarusian National Technical University as Energy Engineer. Studied in RS School in 2022 year but was drafted into the army to pay his debt to the homeland and now he is again in the RS School to finish what was started and get more experience and knowledge.
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
        >
          <img
            alt="Zdanevich"
            src="https://avatars.githubusercontent.com/u/84130329?v=4"
            style="width: 200px; height: 200px;"
          />
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
          >
            Anatoli Zdanevich
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Github: 
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11iau0r-MuiTypography-root-MuiLink-root"
            href="https://github.com/zdan3v1ch"
          >
            https://github.com/zdan3v1ch
          </a>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Role: 
          </p>
          The right hand of the leader
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Significant contributions: 
          </p>
          This guy is a guru of commercetools API. The average user won't see his work, only the one who familiar with Network tab in Devtools. All API requests are his handiwork.
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             BIO: 
          </p>
          Graduated from college, university, and also underwent retraining in “Web Design and Computer Graphics” at the Business Institute of BSU. Completed JS/FE preschool 2022Q2. And now he is again in the RS School in the main course to get more experience and knowledge.
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
        >
          <img
            alt="Schepalina"
            src="https://avatars.githubusercontent.com/u/112850187?v=4"
            style="width: 200px; height: 200px;"
          />
        </div>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-tjvwag-MuiTypography-root"
          >
            Viktoriya Schepalina
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Github: 
          </p>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-11iau0r-MuiTypography-root-MuiLink-root"
            href="https://github.com/90Viktoriya"
          >
            https://github.com/90Viktoriya
          </a>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Role: 
          </p>
          The left hand of the leader
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             Significant contributions: 
          </p>
          This girl is responsible for periodic reminders and checks that the work is in full swing. If you have any problems with login, registration or user profile pages, it's also to her.
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
             BIO: 
          </p>
          Graduated from Surgut State University as specialist in Automated System of information processing and control. Worked as system administrator of SAP system for many years. Forgot all during parental leave. And now she is in RS School to get more experience and knowledge.
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
